### DISCUSSION TITLE
vdj, adata = ddl.pp.check_contigs(vdj, adata) not working

### DISCUSSION THREAD

**Discussion by marius-mathies** [tuonglab/dandelion#406] [Q&A]
I am using output from the 10x cell ranger multi command, and everything is working until I run 
vdj, adata = ddl.pp.check_contigs(vdj, adata) 

Could there be any issue with the pipeline?

Preparing data: 827it [00:00, 2803.68it/s]
Scanning for poor quality/ambiguous contigs: 100%|██████████| 483/483 [00:02<00:00, 189.40it/s]
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[102], line 1
----> 1 vdj, adata = ddl.pp.check_contigs(vdj, adata)

File ~/.conda/envs/conda_python_3.11/lib/python3.11/site-packages/dandelion/preprocessing/_preprocessing.py:5258, in check_contigs(data, adata, productive_only, library_type, umi_foldchange_cutoff, filter_missing, filter_extra, save, verbose, **kwargs)
   5255     dat = dat[dat["cell_id"].isin(adata_.obs_names)].copy()
   5257 if dat.shape[0] == 0:
-> 5258     raise IndexError(
   5259         "No contigs passed filtering. Are you sure that the cell barcodes are matching?"
   5260     )
   5261 if os.path.isfile(str(data)):
   5262     data_path = Path(data)

IndexError: No contigs passed filtering. Are you sure that the cell barcodes are matching?

**Comment by zktuong** [comment]
hi, can you check that the format of your `adata.obs_names` and `vdj.metadata_names` is the same?

**Reply by marius-mathies** [reply]
Thank you for your quick reply, indeed it seems not to match. Would you know a way to circumvent this?

<img width="1128" alt="image" src="https://github.com/user-attachments/assets/b227d02f-e61e-4a42-a579-d7e3756d20e4">

**Reply by marius-mathies** [reply]
I did it like this :)


adata_list = []
for sample in samples:
    adata = sc.read_10x_h5(
        "/n/data2/dfci/pedonc/knoechel/marius/masters_thesis/KarMMa/10X_pipeline/10X_output/02_multi_output/" + sample + "/outs/per_sample_outs/" + sample + "/count/sample_feature_bc_matrix.h5",
        gex_only=True,
    )
    adata.obs["sampleid"] = sample
    # rename cells to sample id + barcode, and cleave the trailing -#
    adata.obs_names = [
        str(j).split("-")[0] + "_nan" for j in adata.obs_names
    ]
    adata.var_names_make_unique()
    adata_list.append(adata)
# no need for index_unique as we already made barcodes unique by prepending the sample ID
adata = adata_list[0].concatenate(adata_list[1:], index_unique=None)

**Reply by zktuong** [reply]
Hmm looks like the way you are amending the cell barcodes are not consistent between your adata object and the vdj data. You should deal with them the same way. For the vdj data, this needs to be dealt with in the vdj.data.cell_id column. I can try to provide some code tomorrow as i'm travelling at the moment. As a start you can look at the advance tutorial on the documentation and see if you can spot how cell barcodes are altered in dandelion.
